packages/dev/gui/src/2D/controls/image.ts

```diff
@@ -555,15 +558,30 @@ export class Image extends Control {
             const cachedData = Image.SourceImgCache.get(value)!;
             this._domImage = cachedData.img;
             cachedData.timesUsed += 1;
-            this._onImageLoaded();
+            if (cachedData.loaded) {
+                this._onImageLoaded();
+            } else {
+                cachedData.waitingForLoad.push(this);
+            }
             return;
         }
         this._domImage = engine.createCanvasImage();
         if (value) {
-            Image.SourceImgCache.set(value, { img: this._domImage, timesUsed: 1 });
+            Image.SourceImgCache.set(value, { img: this._domImage, timesUsed: 1, loaded: false, waitingForLoad: [this] });

```

Comment:
why is the array needed?

Reply:
There might be multiple images that access _domImage while it is trying to load, so they should have their own _onImageLoaded methods when the onload call resolves.

Reply:
Got it! makes sense to keep the reference to all functions, but is there a need for the entire class to be passed? Why not use an observable that is being notified when the dom image was loaded? In this case using the `addOnce` would make sense to make sure the observable doesn't keep unneeded references.

Reply:
Oooh, that makes sense and saves space on the cache, will do it

Reply:
@RaananW done!